s3prl/downstream/runner.py

```diff
@@ -448,64 +496,76 @@ def inference(self):
             self.downstream.model.inference(features, [filename])
 
     def push_to_huggingface_hub(self):
+        """Creates a downstream repository on the Hub and pushes training artifacts to it."""
         if self.args.hf_hub_org.lower() != "none":
             organization = self.args.hf_hub_org
         else:
             organization = os.environ.get("HF_USERNAME")
         huggingface_token = HfFolder.get_token()
         print(f"[Runner] - Organisation to push fine-tuned model to: {organization}")
         
-        # Create repo on the Hub
-        upstream_model = self.args.upstream.replace("/", "__")
-        repo_name = f"superb-s3prl-{upstream_model}-{self.args.downstream}-{str(uuid.uuid4())[:8]}"
+        # Extract upstream repository metadata
+        if self.args.hub == "huggingface":
+            model_info = HfApi().model_info(self.args.upstream, token=huggingface_token)
+            downstream_model_id = model_info.sha
+            # Exclude "/" characters from downstream repo ID
+            upstream_model_id = model_info.modelId.replace("/", "__")
+        else:
+            upstream_model_id = self.args.upstream.replace("/", "__")
+            downstream_model_id = str(uuid.uuid4())[:8]
+        repo_name = f"{upstream_model_id}__{downstream_model_id}"
+        # Create downstream repo on the Hub
         repo_url = HfApi().create_repo(
             token=huggingface_token,
             name=repo_name,
             organization=organization,
             exist_ok=True,
-            private=True,
+            private=False,
         )
         print(f"[Runner] - Created Hub repo: {repo_url}")
 
         # Download repo and copy templates
         HF_HUB_DIR = "hf_hub"
-        REPO_PATH = os.path.join(self.args.expdir, HF_HUB_DIR, repo_name)
+        REPO_ROOT_DIR = os.path.join(self.args.expdir, HF_HUB_DIR, repo_name)
+        REPO_TASK_DIR = os.path.join(REPO_ROOT_DIR, self.args.downstream, self.args.expname)
```

Comment:
Here is where I've added the experiment name to the path we store all the training artifacts

Reply:
Yes this looks good to me!